dotCMS/src/integration-test/java/com/dotcms/publishing/job/SiteSearchJobImplTest.java

```diff
@@ -448,74 +454,220 @@ public void Test_Incremental_Job_Test_Pages_Are_Found_Create_And_Publish_New_Pag
+                .createTestCaseSiteSearch();
+
+        /*
+
+        /*
+         * Given sceneario: default-language content referenced from a page in default lang only.
+         * DEFAULT_CONTENT_TO_DEFAULT_LANGUAGE=false
+         * DEFAULT_PAGE_TO_DEFAULT_LANGUAGE=true.
+         *
+         * Create a Site-Search run only in second language
+         * Expected: searching for the content in default language should give results. The result should be the page in the default language
+         */
+
+        TestCaseSiteSearch case3 = new TestCaseSiteSearchBuilder()
+                .defaultContentToDefaultLanguage(false)
+                .defaultPageToDefaultLanguage(true)
+                .siteSearchSecondLanguage(true)
+                .createContentInDefaultLanguage(true)
+                .createPageInDefaultLanguage(true)
+                .expectedResultsWhenSearchingContentInDefaultLanguage(false)
+                .createTestCaseSiteSearch();
+
+        /*
+
+
+         * Given sceneario: Content in second language only referenced from a page in default language.
+         * DEFAULT_CONTENT_TO_DEFAULT_LANGUAGE=true.
+         * DEFAULT_PAGE_TO_DEFAULT_LANGUAGE=true.
+         * Create a Site-Search run only in second language.
+         * Expected: searching for content in second language should give results. The result should be the page in the default language
+         */
+
+        TestCaseSiteSearch case4 = new TestCaseSiteSearchBuilder()
+                .defaultContentToDefaultLanguage(true)
+                .defaultPageToDefaultLanguage(true)
+                .siteSearchSecondLanguage(true)
+                .createContentInSecondLanguage(true)
+                .createPageInDefaultLanguage(true)
+                .expectedResultsWhenSearchingContentInSecondLanguage(true)
+                .createTestCaseSiteSearch();
+
+
+        /*
+         * Same as above but with DEFAULT_PAGE_TO_DEFAULT_LANGUAGE=false
+         * Expected: searching for content in second language should give NO results.
+         */
+
+        TestCaseSiteSearch case5 = new TestCaseSiteSearchBuilder()
+                .defaultContentToDefaultLanguage(true)
+                .defaultPageToDefaultLanguage(false)
+                .siteSearchSecondLanguage(true)
+                .createContentInSecondLanguage(true)
+                .createPageInDefaultLanguage(true)
+                .expectedResultsWhenSearchingContentInSecondLanguage(false)
+                .createTestCaseSiteSearch();
+
+        /*
+         * Given sceneario: two-language (default and second) content referenced from a page in default lang only. DEFAULT_PAGE_TO_DEFAULT_LANGUAGE=true.
+         * Create a Site-Search run including both languages
+         * Expected: searching content of either version of the content should give results. The result should be the page in the default language
+         */
+
+        TestCaseSiteSearch case6 = new TestCaseSiteSearchBuilder()
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'case6' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5303548)